lib/command/copy_image_from_upstream.rb

```diff
@@ -29,7 +29,7 @@ def call # rubocop:disable Metrics/MethodLength
       ensure_docker_running!
 
       @upstream = config[:upstream]
-      @upstream_org = config.apps[@upstream.to_sym][:cpln_org] || ENV.fetch("CPLN_ORG_UPSTREAM", nil)
+      @upstream_org = config.find_app_config(@upstream)&.dig(:cpln_org) || ENV.fetch("CPLN_ORG_UPSTREAM", nil)
```

Comment:
Let's say we have a  `dummy-test` app in `controlplane.yml` that has `match_if_app_name_starts_with` set to `true`.

The old method wouldn't work if called with, e.g., `dummy-test-123`, because `config.apps` only contains the apps exactly as they are in `controlplane.yml`, and we have `dummy-test` in `controlplane.yml`, not `dummy-test-123`.

The new method works by iterating through `config.apps` looking for an app that matches `dummy-test-123`, so it will find `dummy-test` and match it.

Reply:
I don't understand why we'd use a starts with for the upstream. That should be a validation error. Agree, @dzirtusss